pkg/measurements/factory.go

```
@@ -111,11 +113,34 @@ func NewMeasurementsFactory(configSpec config.Spec, metadata map[string]any, add
 
 func (msf *MeasurementsFactory) NewMeasurements(jobConfig *config.Job, kubeClientProvider *config.KubeClientProvider) *Measurements {
 	ms := Measurements{
-		MeasurementsMap: make(map[string]Measurement, len(msf.Factories)),
+		MeasurementsMap: make(map[string]Measurement),
 	}
 	clientSet, restConfig := kubeClientProvider.ClientSet(jobConfig.QPS, jobConfig.Burst)
+
+	log.Infof("Initializing measurements for job: %s", jobConfig.Name)
 	for name, factory := range msf.Factories {
 		ms.MeasurementsMap[name] = factory.NewMeasurement(jobConfig, clientSet, restConfig)
+		log.Infof("📈 Measurement: %s (global)", name)
+	}
+	if len(jobConfig.Measurements) > 0 {
+		for _, measurement := range jobConfig.Measurements {
+			// Skip if measurement is already registered through global config
+			if _, exists := ms.MeasurementsMap[measurement.Name]; exists {
+				continue
+			}
```

this silently skips job-specific measurements when a global measurement with the same name exists

not logging or documenting this behaviour could lead to user confusion when they define job-specific measurements that mysteriously don't take effect.